be/src/exec/hdfs_scanner.cpp

```diff
@@ -234,8 +234,8 @@ Status HdfsScanner::open_random_access_file() {
 
         // input_stream = CacheInputStream(input_stream)
         if (_scanner_params.use_block_cache) {
-            _cache_input_stream = std::make_shared<io::CacheInputStream>(input_stream, filename, file_size,
-                                                                         _scanner_params.modification_time);
+            _cache_input_stream = std::make_shared<io::CacheInputStream>(_shared_buffered_input_stream, filename,
```

Comment:
sometimes input stream is not shared buffered input stream.

Reply:
> sometimes input stream is not shared buffered input stream.

We only pass the `SharedBufferInputStream` to `CacheInputStream` in our current code. As we process the overlap memory range between io buffer and shared buffer in `CacheInputStream`, wrapping the `SharedBufferInputStraem` parameter make the code simpler.